src/tracks/playabletrack/wavetrack/ui/WaveTrackAffordanceControls.cpp

```diff
@@ -26,33 +26,72 @@
 #include "../../../../ProjectWindows.h"
 #include "../../../../commands/AudacityCommand.h"
 #include "../../../ui/AffordanceHandle.h"
+#include "../../../ui/TextEditHelper.h"
 #include "WaveTrackView.h"//need only ClipParameters
 
 #include "../../../../ProjectHistory.h"
 #include "../../../../SelectionState.h"
 #include "../../../../RefreshCode.h"
 #include "../../../../Theme.h"
+#include "../../../../images/Cursors.h"
+#include "../../../../HitTestResult.h"
 
 class WaveTrackAffordanceHandle final : public AffordanceHandle
 {
+    std::shared_ptr<WaveClip> mTarget;
+    bool mDoubleClickFlag{ false };
 public:
-    WaveTrackAffordanceHandle(const std::shared_ptr<Track>& track) : AffordanceHandle(track) { }
+    WaveTrackAffordanceHandle(const std::shared_ptr<Track>& track, const std::shared_ptr<WaveClip>& target) 
+        : AffordanceHandle(track), mTarget(target)
+    { }
 
-    UIHandle::Result SelectAt(const TrackPanelMouseEvent& event, AudacityProject* pProject) override
+    Result Click(const TrackPanelMouseEvent& event, AudacityProject* project) override
     {
-        const auto track = std::dynamic_pointer_cast<WaveTrack>(TrackList::Get(*pProject).Lock<Track>(GetTrack()));
-
-        auto& viewInfo = ViewInfo::Get(*pProject);
-
-        auto time = viewInfo.PositionToTime(event.event.m_x, event.rect.x);
-        
-        WaveClip* const selectedClip = track->GetClipAtTime(time);
-        if (selectedClip) {
-            viewInfo.selectedRegion.setTimes(selectedClip->GetOffset(), selectedClip->GetEndTime());
+        mDoubleClickFlag = false;
+        if (event.event.LeftDClick())
+        {
+            //To make sure that Double Click drag/release events are not handled by super class
+            mDoubleClickFlag = true;
+
+            auto& viewInfo = ViewInfo::Get(*project);
+            auto affordanceRect = ClipParameters::GetClipRect(*mTarget.get(), viewInfo, event.rect);
+            if (affordanceRect.Contains(event.event.GetPosition()))
+            {
+                if (auto affordanceControl = std::dynamic_pointer_cast<WaveTrackAffordanceControls>(event.pCell))
+                {
+                    return affordanceControl->StartEditClipName(project) ? 
+                        RefreshCode::RefreshCell : RefreshCode::RefreshNone;
+                }
+            }
+            return RefreshCode::RefreshNone;
         }
+        else
+            return AffordanceHandle::Click(event, project);
+    }
+
+    Result Release(const TrackPanelMouseEvent& event, AudacityProject* project, wxWindow* parent) override
+    {
+        if (mDoubleClickFlag)
+            return RefreshCode::RefreshNone;
```

Comment:
You can simplify.  Double Click can OR the return value with Cancelled and then you don't need mDoubleClickFlag.  See the much repeated comment "Do not start a drag" in the source tree for other examples.

Reply:
Thanks!